app/ide-desktop/lib/client/watch.ts

```diff
@@ -33,40 +35,74 @@ console.log('Cleaning IDE dist directory.')
 await fs.rm(IDE_DIR_PATH, { recursive: true, force: true })
 await fs.mkdir(IDE_DIR_PATH, { recursive: true })
 
-const BOTH_BUNDLES_READY = new Promise<Watches>((resolve, reject) => {
+const ALL_BUNDLES_READY = new Promise<Watches>((resolve, reject) => {
     void (async () => {
         console.log('Bundling client.')
-        const clientBundlerOpts: esbuild.BuildOptions = {
-            ...clientBundler.bundlerOptionsFromEnv(),
-            outdir: path.resolve(IDE_DIR_PATH),
-            watch: {
-                onRebuild(error) {
-                    if (error) {
+        const clientBundlerOpts = clientBundler.bundlerOptionsFromEnv()
+        clientBundlerOpts.outdir = path.resolve(IDE_DIR_PATH)
+        // Eslint is wrong here; this is actually `undefined`.
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        ;(clientBundlerOpts.plugins ??= []).push({
+            name: 'enso-on-rebuild',
+            setup: build => {
+                build.onEnd(result => {
+                    if (result.errors.length) {
                         // We cannot carry on if the client failed to build, because electron executable
                         // would immediately exit with an error.
-                        console.error('Client watch bundle failed:', error)
-                        reject(error)
+                        console.error('Client watch bundle failed:', result.errors[0])
+                        reject(result.errors[0])
                     } else {
                         console.log('Client bundle updated.')
                     }
-                },
+                })
             },
-        }
-        const client = await esbuild.build(clientBundlerOpts)
+        })
+        const clientBuilder = await esbuild.context(clientBundlerOpts)
+        const client = await clientBuilder.rebuild()
         console.log('Result of client bundling: ', client)
+        void clientBuilder.watch()
+
+        console.log('Bundling dashboard.')
+        const dashboardOpts = dashboardBundler.bundleOptions()
+        dashboardOpts.plugins.push({
+            name: 'enso-on-rebuild',
+            setup: build => {
+                build.onEnd(() => {
+                    console.log('Dashboard bundle updated.')
+                })
+            },
+        })
+        dashboardOpts.outdir = path.resolve(IDE_DIR_PATH, 'assets')
+        const dashboardBuilder = await esbuild.context(dashboardOpts)
+        const dashboard = await dashboardBuilder.rebuild()
+        console.log('Result of dashboard bundling: ', dashboard)
+        // We do not need to serve the dashboard as it outputs to the same directory.
+        // It will not rebuild on request, but it is not intended to rebuild on request anyway.
+        // This MUST be called before `builder.watch()` as `tailwind.css` must be generated
+        // before the copy plugin runs.
+        void dashboardBuilder.watch()
 
```

Comment:
As we agreed, dashboard should not be a separate bundle landing in assets, right? Both cloud and desktop versions should just bundle `content` and dashboard should be dependency of content (which needs to probably be bundled before loading to content, but not to "assets", as `content` should be the main entry point always). We agreed to keep it for now this way to have these 2 entry points. However, when do we plan to unify it and use `content` as the main entry point and where is a task for it? We need to fix it, as otherwise we are growing bigger and bigger tech debt here, so I just want to be sure this is planned as soon as possible.

Reply:
As we are not planning to open `https://cloud.enso.org` the separate entrypoint is mainly being used for development and review/debug stuff. 

We are planning to completely remove it as soon as we have pending dashboard PR merged.

Reply:
Here is the task for it https://github.com/enso-org/cloud-v2/issues/345

Reply:
We are not planning to provide users with access to `https://cloud.enso.org` for now, but it still should work. Hopefully sooner than later we will be able to work in the cloud. Anyway, I understand that the plan is that `content` will be used as the main entry point for both electron and the web version right?

Anyway, this build script puts the results into "assets", so it does not look like it's for dev/testing only, as "assets" is bundled with the final app.

Reply:
Anyway, I understand that after the PR you mentioned is merged, we will move it this way that content will be the main entry point for both electron and the web version, is that correct?

Reply:
yes, we need to push dashboard to be ready then we can polish and fix this technology debt